added project carbon_rover

ric96 opened the pull request:
- added source files for project
- added README.md for the project

Signed-off-by: Sahaj Sarup <sahaj.sarup@linaro.org>

ric96 committed:
added project carbon_rover

- added source files for project
- added README.md for the project

Signed-off-by: Sahaj Sarup <sahaj.sarup@linaro.org>

Mani-Sadhasivam commented on carbon_rover/README.md:
```diff
@@ -0,0 +1,134 @@
+- [6x Ultrasonic Sensors HC-SR04](https://www.robomart.com/buy-4-pin-ultrasonic-sensor-module-arduino-raspberry?search=hcsr04&dsearch=hcsr04)
+- 6x 1KOhm Resistors
+- 6x 2KOhm Resistors
+- [4x IR Edge Detection Sensors](https://www.robomart.com/ir-based-digital-color-sensor-black-white)
```
IR sensor quantity is 4 or 6?

ric96 replied:
4, one for each corner

Mani-Sadhasivam commented on carbon_rover/rover/src/main.c:
```diff
@@ -0,0 +1,264 @@
+
+void main(void)
+{
+	gpioa = device_get_binding(PORT);
```
Add error check please

Mani-Sadhasivam reviewed (changes requested):
Added trivial comments. Overall LGTM

ric96 committed:
carbon_rover: source and readme fixes

- Fixed no. of IR Sensors required to 4
- Added error check to GPIO init

Signed-off-by: Sahaj Sarup <sahaj.sarup@linaro.org>

Mani-Sadhasivam commented on carbon_rover/rover/src/main.c:
```diff
@@ -60,36 +60,107 @@ uint32_t get_us(uint32_t trig, uint32_t echo, struct device *dev);
+
 	gpioa = device_get_binding(PORT);
+	if (!gpioa) {
+		printk("Cannot find %s!\n", PORT);
```
We should return here. No point in continuing once the port is not found.

ric96 committed:
carbon_rover: fixed gpio error and thread handling

 - gpio init error now results in a return statement
 - threads are now spawned by main after gpio init
 - added configs to prj.conf

Signed-off-by: Sahaj Sarup <sahaj.sarup@linaro.org>
